### PR TITLE
Improve error message when port or fqbn flags are not set

### PR DESCRIPTION
**Pull request by federicobond**
This changeset marks the fqbn and port CLI flags as required in the compile and upload commands. The error message is thus changed from the generic:
```
Error during Upload: no upload port provided
```
To the more directly actionable:
```
Error: required flag(s) "fqbn", "port" not set
```
Which is followed by the CLI help text.

### CODE REVIEW

**Comment by masci** [comment]
Hi @federicobond thanks for the PR.

The code doesn't compile, can you please go through this doc https://github.com/arduino/arduino-cli/blob/master/CONTRIBUTING.md, make sure tests pass on your computer and then push again?

I'll review your code as soon as the tests pass, thanks!

**Comment by federicobond** [comment]
Hi @masci, sorry about that, I was trying to get the integration tests to pass and I botched the latest code changes! I still have some problems with the test match because the error is written to stderr, which is not captured by `executeWithArgs`. I will try to fix that soon.

**Comment by federicobond** [comment]
@masci done!

File: cli/cli_test.go [hunk 1]
```diff
@@ -42,26 +43,28 @@ var (
...
 	stdOut = os.Stdout
+	stdErr = os.Stderr
 
 	currDownloadDir   string
 	currDataDir       string
 	currSketchbookDir string
 )
 
-type stdOutRedirect struct {
+type outputRedirect struct {
```
Comment (masci) [inline cli/cli_test.go]: :cake:

**Comment by federicobond** [comment]
I just noticed that this does not play along well with the board FQBN configured via the `sketch.json` file. My bad! I will try to submit a fix shortly.

**Pull request merged**